### PR TITLE
Search refactor

### PR DESCRIPTION
**Pull request by microbit-robert**
Move search logic into `search-hooks.tsx` and clear search query and results when language is changed.

### CODE REVIEW

**Comment by microbit-matt-hillsdon** [comment]
This looks good to me now. @microbit-robert, feel free to merge it once you've taken a look at my changes.

**Comment by microbit-robert** [comment]
@microbit-matt-hillsdon There is now a slight delay in closing the search results on clearing the search input, because we no longer directly set results as undefined, and we let the de-bounced search do it for us.

File: src/documentation/search/search-hooks.tsx [hunk 1]
```diff
@@ -22,15 +39,53 @@ export const useSearch = (): Search => {
...
+    [setResults, isUnmounted, logging]
+  );
+
+  const [{ languageId }] = useSettings();
+  useEffect(() => {
+    setQuery("");
+  }, [languageId]);
+
+  useEffect(() => {
+    debouncedSearch(query);
```
Comment (microbit-robert) [inline src/documentation/search/search-hooks.tsx:80]: Something like this prevents the delay in closing the results on clearing the search input.
```suggestion
    if (!query.trim()) {
      setResults(undefined);
    } else {
      debouncedSearch(query);
    }
```

Comment (microbit-robert) [reply src/documentation/search/search-hooks.tsx]: [`[6d15abb](https://github.com/microbit-foundation/python-editor-next/pull/574/commits/6d15abb74b7269c1afd42518e6c0246210169ba1)`](https://github.com/microbit-foundation/python-editor-next/pull/574/commits/6d15abb74b7269c1afd42518e6c0246210169ba1)

**Review by microbit-matt-hillsdon — APPROVED** [review]

**Pull request merged**